### ISSUE TITLE
`yarn run production` fails

### ISSUE THREAD

**Issue by MunifTanjim** [yarnpkg/yarn#2349]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

`package.json` scripts part:

```javascript
  "scripts": {
    "production": "webpack -p"
  },
```

yarn command:

```shell
yarn run production
```

output:

```shell
yarn run v0.18.1
error No command specified.
info Commands available from binary scripts: webpack, webpack-dev-server
info Project commands
   - production
question Which command would you like to run?:
```

**If the current behavior is a bug, please provide the steps to reproduce.**

Same as above.

**N.B.**: The problem seems to exist for only if the script is named `production`. For example, if we change the name:

```javascript
  "scripts": {
    "product": "webpack -p"
  },
```

`yarn run product` works just fine.

**What is the expected behavior?**

run the script with no error...

**Please mention your node.js, yarn and operating system version.**
node.js : v6.9.1
yarn : v0.18.1
os : UbuntuMATE 16.10

**Comment by benmvp** [comment]
FWIW, I was able to reproduce the same issue in v0.18.1. If you do `npm run production` it works just fine. Maybe `production` is some sort of secret reserved word...

**Comment by jerguslejko** [comment]
+1 same here

**Comment by iredchuk** [comment]
This should be fixed in 0.19.
See #2112
